static/src/javascripts/projects/common/modules/navigation/newSticky.js

```diff
@@ -0,0 +1,90 @@
+define([
+    'common/utils/fastdom-promise',
+    'Promise',
+    'common/utils/$',
+    'common/utils/mediator',
+    'common/utils/detect'
+], function (
+    fastdom,
+    Promise,
+    $,
+    mediator,
+    detect) {
+
+    return function () {
+        if (detect.getBreakpoint() !== 'mobile') {
+            fastdom.read(function () {
```

Comment:
promises don't create threads no, but `fastdom` batches reads and writes in a single `requestAnimationFrame` call, and there are many of them throughout the code, so it's best to keep them as short as possible.